ddtrace/sampler.py

```diff
@@ -150,32 +149,29 @@ def set_sample_rate(
         # type: (...) -> None
         self._by_service_samplers[self._key(service, env)] = RateSampler(sample_rate)
 
-    def _set_priority(self, span, priority):
-        # type: (Span, int) -> None
-        span.context.sampling_priority = priority
-        span.sampled = priority > 0  # Positive priorities mean it was kept
-
-    def _set_sampler_decision(self, span, sampler, sampled):
-        # type: (Span, RateSampler, bool) -> None
-        priority = AUTO_KEEP if sampled else AUTO_REJECT
-        self._set_priority(span, priority)
-        if sampler is self._default_sampler:
-            mechanism = SamplingMechanism.DEFAULT
-        else:
-            mechanism = SamplingMechanism.AGENT_RATE
-            span.set_metric(SAMPLING_AGENT_DECISION, sampler.sample_rate)
-        set_sampling_decision_maker(span.context, mechanism)
+    def sample(self, span, allow_false=False):
+        sampled, sampler = self._make_sampling_decision(span)
+        _set_sampling_tags(
+            span,
+            sampled,
+            sampler.sample_rate,
+            self._choose_priority_category(sampler is self._default_sampler),
+        )
+        return not allow_false or sampled
```

Comment:
ah it's because `not False or True` will return True. I was thinking or only cared about if the value for the first object was `None`